release.py

```diff
@@ -80,9 +80,9 @@ def fail(msg):
 
     if delete_gitrefs:
         try:
-            cmd("Resetting repository working state to branch %s" % starting_branch, "git reset --hard HEAD && git checkout %s" % starting_branch, shell=True)
-            cmd("Deleting git branches %s" % release_version, "git branch -D %s" % release_version, shell=True)
-            cmd("Deleting git tag %s" %rc_tag , "git tag -d %s" % rc_tag, shell=True)
+            cmd("Resetting repository working state to branch %s" % starting_branch, "git reset --hard HEAD && git checkout %s" % starting_branch, shell=True, allow_failure=True)
+            cmd("Deleting git branches %s" % release_version, "git branch -D %s" % release_version, shell=True, allow_failure=True)
+            cmd("Deleting git tag %s" %rc_tag , "git tag -d %s" % rc_tag, shell=True, allow_failure=True)
```

Comment:
If the first line fails which is correctly switching the branch, we will proceed with next line and end up deleting an incorrect branch. We don't want that. Am I missing something?